src/time/Storage.cpp

```diff
@@ -148,7 +163,19 @@ Eigen::VectorXd Storage::sample(double normalizedDt) const
 
   auto data = getTimesAndValues();
 
-  return math::bspline::interpolateAt(data.first, data.second, usedDegree, normalizedDt);
+  //If t equals to one timestep in data.first then we want to return the corresponding sample
+  for (int i = 0; i < data.first.size(); i++) {
+    if (math::equals(data.first[i], normalizedDt)) {
+      return data.second.col(i);
+    }
+  }
+
+  //Create a new bspline if _bspline does not already contain a spline
+  if (!_bspline.has_value()) {
+    _bspline = std::make_optional<math::Bspline>(data.first, data.second, usedDegree);
```

Comment:
At this point, we know that the bspline is empty. So we can directly construct the B-Spline inside the optional using [`.emplace()`](https://en.cppreference.com/w/cpp/utility/optional/emplace).

```suggestion
    _bspline.emplace(data.first, data.second, usedDegree);
```